modules/javafx.graphics/src/main/java/com/sun/javafx/iio/javax/XImageLoader.java

```diff
@@ -196,14 +199,23 @@ public ImageFrame load(int imageIndex, double w, double h, boolean preserveAspec
                 var colorModel = (IndexColorModel)image.getColorModel();
                 var palette = new int[colorModel.getMapSize()];
                 colorModel.getRGBs(palette);
+
+                var imageType = colorModel.hasAlpha()
+                    ? colorModel.isAlphaPremultiplied()
+                        ? ImageStorage.ImageType.PALETTE_ALPHA_PRE
+                        : ImageStorage.ImageType.PALETTE_ALPHA
+                    : ImageStorage.ImageType.PALETTE;
+
+                var scanlineStride = switch(image.getSampleModel()) {
```

Comment:
Just a general comment here, on the use of `var`; in my opinion, `var` helps the writer of the code, but almost never helps the reader of the code who now must read the RHS and manually determine what the type is.  As we should be aiming for maintainable code and code that's as easy to read as possible without having to guess (there's usually enough guessing involved already), I always find it hard how the use of `var` can be justified anywhere. 

Also in my experience, `var` complicates refactors as it will morph with the target type, making all refactor errors occur at the use location instead of the declaration site when `var` isn't used (ie. a refactor could indicate dozens of errors, while fixing 2 declarations could solve all of them).

So why make us guess that `scanlineStride` is an `int` here?  And why force us to read the right hand sides of `colorModel`, `palette` and `imageType`...

Reply:
`scanlineStride` is useful for all the `ImageFrame` constructors, or do you think recomputing it (with the risk of it being different) is still the best course of action given that AWT supplies it and might be using some other alignment/padding?

Reply:
> `scanlineStride` is useful for all the `ImageFrame` constructors, or do you think recomputing it (with the risk of it being different) is still the best course of action given that AWT supplies it and might be using some other alignment/padding?

I've changed it to always take the scanline stride from AWT and never recompute. Note that AWT measures the stride in "data elements", not in bytes. I've marked the points in the source code where the meaning of stride changes from "data elements" to "bytes".